homeassistant/components/pjlink/media_player.py

```diff
@@ -85,50 +89,73 @@ def __init__(self, host, port, name, encoding, password):
         self._muted = False
         self._pwstate = MediaPlayerState.OFF
         self._current_source = None
-        with self.projector() as projector:
-            if not self._name:
-                self._name = projector.get_name()
-            inputs = projector.get_inputs()
-        self._source_name_mapping = {format_input_source(*x): x for x in inputs}
-        self._source_list = sorted(self._source_name_mapping.keys())
+        self._source_name_mapping = {}
+        self._source_list = []
+        self._available = self._initialize()
+
+    def _force_off(self):
+        self._pwstate = MediaPlayerState.OFF
+        self._muted = False
+        self._current_source = None
+
+    def _initialize(self):
+        try:
+            with self.projector() as projector:
+                if not self._name:
+                    self._name = projector.get_name()
+                inputs = projector.get_inputs()
+                self._source_name_mapping = {format_input_source(*x): x for x in inputs}
+                self._source_list = sorted(self._source_name_mapping)
+                return True
+        except ProjectorError as err:
+            if str(err) == ERR_PROJECTOR_UNAVAILABLE:
+                return False
+            raise
 
     def projector(self):
         """Create PJLink Projector instance."""
 
-        projector = Projector.from_address(
-            self._host, self._port, self._encoding, DEFAULT_TIMEOUT
-        )
-        projector.authenticate(self._password)
-        return projector
+        try:
+            projector = Projector.from_address(
+                self._host, self._port, self._encoding, DEFAULT_TIMEOUT
+            )
+            projector.authenticate(self._password)
+            return projector
+        except (socket.timeout, OSError) as err:
```

Comment:
Preferably the library should be updated to catch these errors and raise a library specific error similar to `ProjectorError`. Please try to extend the library with this in a future follow up.

Reply:
The original library does not seem to be maintained any longer. I'll see if I can find a replacement fork or start one myself.